### PR TITLE
Add support for binning by day to histogram tile + visual adjustments

### PR DESCRIPTION
**Pull request by juliawu**
This PR makes the following changes to the histogram tile:
* Adds support for binning data by day
* Bins by day for "last 30 days" and "YYYY-MM" in date dropdowns
* Adjusts x-axis so first labels are no longer cut off
* Adds a maximum bar width so we avoid extremely wide singleton bars

Screenshots:

Screenshot 2023-02-13 at 1 47 59 PM
Screenshot 2023-02-13 at 1 48 21 PM
Screenshot 2023-02-13 at 1 48 41 PM

### CODE REVIEW

File: static/js/components/tiles/histogram_tile.tsx [hunk 1]
```diff
@@ -42,6 +42,13 @@ interface HistogramTilePropType {
   title: string;
 }
 
+/**
```
[the remark this answers is not in the document]
Comment (juliawu) [reply static/js/components/tiles/histogram_tile.tsx]: Good idea. Done!

File: static/js/components/tiles/histogram_tile.tsx [hunk 2]
```diff
@@ -95,6 +134,58 @@ function getMonthsArray(dateSetting: string): string[] {
   return months;
 }
 
+/**
+ * Helper function to bin data into daily bins
+ */
+function binDataByDay(
```
[the remark this answers is not in the document]
Comment (juliawu) [reply static/js/components/tiles/histogram_tile.tsx]: Good idea, done!

File: static/js/components/tiles/histogram_tile.tsx [hunk 3]
```diff
@@ -95,6 +134,58 @@ function getMonthsArray(dateSetting: string): string[] {
...
+
+  // Initialize all bins at zero
+  const daysToPlot = getDaysArray(dateSetting);
+  for (const day of daysToPlot) {
+    bins.set(day, 0);
+  }
+
+  for (const event of disasterEventPoints) {
+    // Get start time in YYYY-MM-DD
+    const eventDay = event.startDate.slice(0, "YYYY-MM-DD".length);
```
[the remark this answers is not in the document]
Comment (juliawu) [reply static/js/components/tiles/histogram_tile.tsx]: done

File: static/js/components/tiles/histogram_tile.tsx [hunk 4]
```diff
@@ -210,10 +311,19 @@ export function HistogramTile(props: HistogramTilePropType): JSX.Element {
...
     setHistogramData: (data: DataPoint[]) => void
   ): void {
-    // TODO(juliawu): detect whether to bin by month or day depending on the
-    //                user's selected time setting
-    // TODO(juliawu): add 'last30days' case handling
-    const histogramData = binDataByMonth(disasterEventPoints, dateSetting);
+    let histogramData = [];
+    // Try binning by day if dateSetting is "last 30 days" or a month
+    if (
+      dateSetting == "lastThirtyDays" ||
```
[the remark this answers is not in the document]
Comment (juliawu) [reply static/js/components/tiles/histogram_tile.tsx]: good catch. done!

File: static/js/components/tiles/histogram_tile.tsx [hunk 5]
```diff
@@ -63,6 +70,38 @@ function getLastDayOfMonth(dateString: string): string {
...
+  if (startDate.length == "YYYY".length && endDate.length == "YYYY".length) {
+    startDate = `${startDate}-01-01`;
+    endDate = `${endDate}-12-31`;
+  } else if (
+    startDate.length == "YYYY-MM".length &&
+    endDate.length == "YYYY-MM".length
+  ) {
+    startDate = `${startDate}-01`;
+    endDate = `${getLastDayOfMonth(endDate)}`;
+  }
```
[the remark this answers is not in the document]
Comment (juliawu) [reply static/js/components/tiles/histogram_tile.tsx:96]: created a helper function

File: static/js/components/tiles/histogram_tile.tsx [hunk 6]
```diff
@@ -64,70 +72,123 @@ function getLastDayOfMonth(dateString: string): string {
 }
 
 /**
- * Helper function for getting all months along x-axis to display.
- * Used for initializing bins when binning monthly.
+ * Get the earliest and latest date in YYYY-MM-DD to show data for
```
Comment (chejennifer) [inline static/js/components/tiles/histogram_tile.tsx:75]: nit: add a period at the end of this sentence

File: static/js/components/tiles/histogram_tile.tsx [hunk 7]
```diff
@@ -64,70 +72,123 @@ function getLastDayOfMonth(dateString: string): string {
...
-  dateSetting: string
+  dateSetting: string,
+  format: string
 ): DataPoint[] {
   if (_.isEmpty(disasterEventPoints)) {
     return [];
   }
 
-  // Track YYYY-MM -> number of events
+  // Track YYYY-MM / YYYY-MM-DD -> number of events
```
Comment (chejennifer) [inline static/js/components/tiles/histogram_tile.tsx:157]: nit: just "Track date -> number of events"

**Review by chejennifer — APPROVED** [review]
thanks for the changes! Just small comment nits

File: static/js/components/tiles/histogram_tile.tsx (hunk 6 above)
Comment (juliawu) [reply static/js/components/tiles/histogram_tile.tsx]: Added!

File: static/js/components/tiles/histogram_tile.tsx (hunk 7 above)
Comment (juliawu) [reply static/js/components/tiles/histogram_tile.tsx]: changed to "date"

**Pull request merged**